gio/src/subclass/initable.rs

```diff
@@ -0,0 +1,177 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::object::Cast;
+use glib::translate::*;
+use glib::Error;
+
+use glib::subclass::prelude::*;
+
+use std::ptr;
+
+use crate::Cancellable;
+use crate::Initable;
+
+pub trait InitableImpl: ObjectImpl + Send {
+    fn init(&self, initable: &Self::Type, cancellable: Option<&Cancellable>) -> Result<(), Error>;
+}
+
+pub trait InitableImplExt: ObjectSubclass {
```

Comment:
I don't get where this extension trait is used

Reply:
I don't either, I just followed the standard interface template (mostly taking "inspiration" from Seekable.rs).

I will ask in a comment in the pr to the upstream project.

Reply:
Reading it twice - I think it's meant to be used by objects subclassing an object which in turns implements an interface, to call the subclassed object interface method from the subclassing one.

In Java/C# pseudocode which is easier:


```
public interface IMyInterface {
     void some_method();
}

public class MyBaseObject implements IMyInterface {
     void some_method() { ... }
}

public class MyObject extends MyBaseObject {
     override void some_method() {
         // to do this
         base.some_method();
     }
}
```